presto-main/src/test/java/com/facebook/presto/operator/aggregation/fixedhistogram/TestFixedDoubleBreakdownHistogram.java

```diff
@@ -0,0 +1,173 @@
+{
+    @Test
+    public void testGetters()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        assertEquals(histogram.getBucketCount(), 200);
+        assertEquals(histogram.getMin(), 3.0);
+        assertEquals(histogram.getMax(), 4.0);
+    }
+
+    @Test
+    public void testIllegalBucketWeightCount()
+    {
+        try {
+            new FixedDoubleBreakdownHistogram(-200, 3.0, 4.0);
+            fail("exception expected");
+        }
+        catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("bucketCount"));
+        }
+    }
+
+    @Test
+    public void testIllegalMinMax()
+    {
+        try {
+            new FixedDoubleBreakdownHistogram(-200, 3.0, 3.0);
+            fail("exception expected");
+        }
+        catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("bucketCount"));
+        }
+    }
+
+    @Test
+    public void testBasicOps()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        histogram.add(3.1, 100.0);
+        histogram.add(3.8, 200.0);
+        histogram.add(3.1, 100.0);
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToDouble(c -> c.getWeight()).sum(),
+                300.0);
+    }
+
+    @Test
+    public void testEqualValuesDifferentWeights()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        histogram.add(3.5, 0.2, 1);
+        histogram.add(3.5, 0.4, 1);
+        histogram.add(3.5, 0.3, 2);
+        Streams.stream(histogram.iterator()).forEach(c -> {
+            assertTrue(c.getLeft() <= 3.5);
+            assertTrue(c.getRight() >= 3.5);
+            assertTrue(c.getLeft() < c.getRight());
+        });
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToLong(c -> 1).sum(),
+                3);
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToDouble(c -> c.getWeight()).sum(),
+                0.9);
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToLong(c -> c.getCount()).sum(),
+                4);
+    }
+
+    @Test
+    public void testMassive()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(100, 0.0, 1.0);
+
+        ArrayList<Double> values = new ArrayList<>();
+        ArrayList<Double> weights = new ArrayList<>();
+
+        IntStream.range(0, 1000000).forEach(i -> {
+            double value = Math.random();
+            double weight = ((int) (10 * Math.random())) / 10.0;
+            values.add(value);
+            weights.add(weight);
+            histogram.add(value, weight, 1);
+        });
+
+        Streams.stream(histogram.iterator()).forEach(b -> {
```

Comment:
instead of nested stream, just use nested for-loop

Reply:
Again, not sure why. Could you elaborate? Thanks!

Reply:
@atavory : For simple operations, stream and lambda-style coding makes code concise and more "functional programming". However, when the inner logic is complicated (and even includes nested loop), it defeat the purpose of using lambda. And plain for-loop is easy to read.

Reply:
One rule of thumb is to ask: is this necessary? In theory all for-loop can be replaced with `IntStream`. But we don't always do that, right?